src/core/Object3D.js

```diff
@@ -1015,6 +1015,19 @@ class Object3D extends EventDispatcher {
 
 	}
 
+	dispose() {
+
+		while ( this.children.length > 0 ) {
+
+			const child = this.children[ 0 ];
+			child.dispose();
```

Comment:
Same with textures. Only because a 3D object is removed, it does not _always_ mean its children are removed as well.

Reply:
The point is to clear all Object's memory, like a `Scene`. If assets are shared, why not set to null before calling `dispose`?

Reply:
There is no point in calling `dispose()` on `Object3D` level if you can't call `dispose()` on materials and geometries.